src/main/java/net/dongliu/apk/parser/struct/resource/TypeExtensions.kt

```diff
@@ -0,0 +1,136 @@
+package net.dongliu.apk.parser.struct.resource
+
+
+fun Type.dirName(): String {
+    val dirBuilder = StringBuilder("/${this.name}")
+
+    if (this.resTableConfig.mcc > 0) {
+        dirBuilder.append("-mcc${String.format("%03d", this.resTableConfig.mcc)}")
+    }
+    if (this.resTableConfig.mnc > 0) {
+        dirBuilder.append("-mnc${String.format("%03d", this.resTableConfig.mnc)}")
+    }
+
+    if (!this.locale.language.isNullOrEmpty()) {
+        dirBuilder.append("-${this.locale.language}")
+    }
+    if (!this.locale.country.isNullOrEmpty()) {
+        dirBuilder.append("-r${this.locale.country}")
+    }
+
+    dirBuilder.append(
+        when (this.resTableConfig.screenLayout.toInt() and 0x000000c0) {
```

Comment:
Where did `0x000000c0` and other bitmasks below come from? Are they documented anywhere?

Reply:
The `ResourceTableParser` class has a comment with the links to the documentation

Reply:
I added two extra doc links there